src/rtapi/uspace_rtapi_app.cc

```diff
@@ -919,6 +922,7 @@ int Posix::task_delete(int id)
 static int find_rt_cpu_number() {
     if(getenv("RTAPI_CPU_NUMBER")) return atoi(getenv("RTAPI_CPU_NUMBER"));
 
+#ifdef __linux__
     cpu_set_t cpuset_orig;
     int r = sched_getaffinity(getpid(), sizeof(cpuset_orig), &cpuset_orig);
```

Comment:
FreeBSD does have pthread_setaffinity_np(), but it lacks sched_getaffinity().  Also, the type of one of the arguments to pthread_setaffinity_np() is different.